standards/naming.md

```diff
@@ -118,56 +118,119 @@ collectedItem: []
         "publisherName": "My Publishing House"  // this name attribute MUST have a prefix for another entity, since its not the name of this entity "author"
-        "agentId": 2                            // this name is appropriate with Id following its associated entity name.
     }
 }
  
 // CORRECT EXAMPLE
-GET /v1/books/2345
+GET /v1/books/5196ab21
 RESPONSE
 {
-    "id": 2345,                            
+    "id": "5196ab21",                            
     "name": "The Best Book",
     "author": {                    
-        "id":456,                      
+        "id": "3793213e",                      
         "name": "John Doe",            
-        "publisherName": "My Publishing House",
-        "agentId": 2
+        "publisherName": "My Publishing House"
     }
 }
 ```
 
+## Identifiers
+
+- An entity with a unique identifier **MUST** be labeled as `id`.
+
+```
+// CORRECT
+{
+    "id": "b96cb3ead9a9",                            
+    "name": "The Best Book"
+}
+```
+
+- Any unique identifier of an entity **MUST** have Id after the entity name when referenced from another entity.
+
+```
+// CORRECT
+{
+    "id": "b96cb3ead9a9",
+    "name": "The Best Book",
+    "authorId": "3793213e"      // usage of "authorId" references this is the id for a different entity.
+}
+```
+
+- Unique identifiers **SHOULD** be string values and considered opaque as described in [serialization of numbers](serialization.md#number).
+
+```
+// INCORRECT
+{
+    "id": 2
+}
+
+// CORRECT
+{
+    "id": "b96cb3ead9a9"
+}
+```
+
+## Domain References
+
+- Unique references to other significant entities within your API Domain, but in a different root resource **SHOULD** use [URN (Uniform Resource Name)](https://en.wikipedia.org/wiki/Uniform_Resource_Name)-like references where appropriate.  
+    - URN-like refers to using the convention of specifying agreed upon namespaces, entities followed by the object identifier, but without the `urn` prefix or formal registration of the namespace ID with IANA according to [RFC8141](https://datatracker.ietf.org/doc/rfc8141/). This provides interoperability and durability benefits within your API and endpoint ecosystem.
+    - URN-like values **MUST** the form of standard `URNs` without the `urn:` prefix to avoid confusion of official registration. A simplified example would be: `{namespace}:{entity}:{id}` (e.g. `spsc:book:b96cb3ead9a9`).
+        - URN-like values **MUST** have a max-length of 255 characters.
+        - URN-like values **MUST** be case-sensitive.
+        - URN-like `{namespace}` **MUST** be a single value applied across all API endpoint response for the usage of self-referencing `ref` properties.
+        - URN-like `{namespace}` **MUST** only contain lowercase alpha characters `[a-z]` with a maximum length of 10 characters.
+        - URN-like `{entity}` **MUST** only contain lowercase alpha characters `[a-z]` with a maximum length of 20 characters.
```

Comment:
Maybe Alex's comment was directed, here?  (i.e. entity vs. namespace which is just staticly `spsc`... right?)

Reply:
Updated the explicit usage of `spsc` within sps commerce.